rad/src/lib.rs

```diff
@@ -39,26 +40,37 @@ pub struct RADRequestExecutionReport {
     pub tally: RadonReport<RadonTypes>,
 }
 
+/// Executes a data request locally.
+/// The `inputs_injection` allows for disabling the actual retrieval of the data sources and
+/// the provided strings will be fed to the retrieval scripts instead. It is therefore expected that
+/// the length of `sources_injection` matches that of `request.retrieve`.
 pub async fn try_data_request(
     request: &RADRequest,
     settings: RadonScriptExecutionSettings,
     inputs_injection: Option<&[&str]>,
 ) -> RADRequestExecutionReport {
     let context = &mut ReportContext::default();
 
-    let mut retrieve_responses = Vec::new();
+    let retrieve_responses = if let Some(inputs) = inputs_injection {
+        assert_eq!(inputs.len(), request.retrieve.len(), "Tried to locally run a data request with a number of injected sources different than the number of retrieval paths ({} != {})", inputs.len(), request.retrieve.len());
 
-    if let Some(inputs) = inputs_injection {
+        let mut retrieve_responses = Vec::new();
         for (retrieve, input) in request.retrieve.iter().zip(inputs.iter()) {
             retrieve_responses.push(run_retrieval_with_data_report(
                 retrieve, input, context, settings,
             ))
         }
+
+        retrieve_responses
     } else {
-        for retrieve in &request.retrieve {
-            retrieve_responses.push(run_retrieval_report(retrieve, settings).await)
-        }
-    }
+        block_on(join_all(
```

Comment:
Using a blocking function inside an async context, this used to panic at runtime.

But we can as remove the `async` keyword from `pub async fn` and it will should work fine.

Reply:
Oh wait, this has an easier solution, just use `join_all(...).await`